src/dso_api/dynamic_api/remote/clients.py

```diff
@@ -42,6 +42,109 @@ class RemoteResponse:
     data: Union[list, dict]
 
 
+def call(pool: urllib3.PoolManager, url: str, **kwargs) -> HTTPResponse:
+    """Make an HTTP GET call. kwargs are passed to pool.request."""
+
+    host = urlparse(url).netloc
+    t0 = time.perf_counter_ns()
+    try:
+        # Using urllib directly instead of requests for performance
+        response: HTTPResponse = pool.request(
+            "GET",
+            url,
+            timeout=60,
+            retries=False,
+            **kwargs,
+        )
+    except (TimeoutError, urllib3.exceptions.TimeoutError) as e:
+        # Socket timeout
+        logger.error("Proxy call to %s failed, timeout from remote server: %s", host, e)
+        raise GatewayTimeout() from e
+    except (OSError, urllib3.exceptions.HTTPError) as e:
+        # Socket connect / SSL error (HTTPError is the base class for errors)
+        logger.error("Proxy call to %s failed, error when connecting to server: %s", host, e)
+        raise ServiceUnavailable(str(e)) from e
+
+    # Log response and timing results
+    level = logging.ERROR if response.status >= 400 else logging.INFO
+    logger.log(
+        level,
+        "Proxy call to %s, status %s: %s, took: %.3fs",
+        host,
+        response.status,
+        response.reason,
+        (time.perf_counter_ns() - t0) * 1e-9,
+    )
+
+    if response.status >= 200 and response.status < 300:
+        return response
+
+    # We got an error.
+    if logger.isEnabledFor(logging.DEBUG):
```

Comment:
Is the `if` necessary? Because the logger handles this for you depending on the log level.

Reply:
This is ancient code from 2020, see ab346a13764c31e6b804f3b8d11708708b079f31. It looks overly cautious to me too, I'll see if I can get rid of it.

Reply:
I think the issue is that `.data` is a `property` that parses and caches the response, which we don't want to do except if we actually want to print the message.

Reply:
Ah, makes sense.